plugin.json

```diff
@@ -19,8 +19,77 @@
         "bundle_path": "webapp/dist/main.js"
     },
     "settings_schema": {
-        "settings": [],
         "header": "Configure the Content Moderator plugin.",
-        "footer": "* To report an issue, make a suggestion, or contribute, [check the repository](https://github.com/mattermost/mattermost-plugin-content-moderator)."
+        "footer": "* To report an issue, make a suggestion, or contribute, [check the repository](https://github.com/mattermost/mattermost-plugin-content-moderator).",
+        "settings": [
+            {
+                "key": "enabled",
+                "display_name": "Enable Content Moderation",
+                "type": "bool",
+                "help_text": "When true, content moderation is enabled.",
+                "default": true
+            },
+            {
+                "key": "type",
+                "display_name": "Moderation Provider",
+                "type": "dropdown",
+                "help_text": "Select which content moderation provider to use.",
+                "default": "azure",
+                "options": [
+                    {
+                        "display_name": "Azure AI Content Safety",
+                        "value": "azure"
+                    }
+                ]
+            },
+            {
+                "key": "endpoint",
+                "display_name": "API Endpoint",
+                "type": "text",
+                "help_text": "The endpoint URL for the moderation API.",
+                "placeholder": "https://your-resource.cognitiveservices.azure.com/"
+            },
+            {
+                "key": "apiKey",
+                "display_name": "API Key",
+                "type": "text",
+                "help_text": "Your moderation API key.",
+                "placeholder": "Enter your API key here"
+            },
+            {
+                "key": "moderateAllUsers",
+                "display_name": "Moderate All Users",
+                "type": "bool",
+                "help_text": "When true, content from all users will be moderated. When false, only content from specified users will be moderated.",
+                "default": false
+            },
+            {
+                "key": "moderationTargets",
+                "display_name": "Moderation Targets",
+                "type": "custom",
+                "help_text": "If not moderating all users, the plugin moderates content from these User IDs."
+            },
+            {
+                "key": "threshold",
+                "display_name": "Content Moderation Threshold",
+                "type": "dropdown",
+                "help_text": "Severity threshold for all content categories (Low filters most aggressively).",
```

Comment:
Is this standard across other providers outside of azure?

Reply:
No, not really. Some providers give a binary `flagged` or `!flagged` result, but many will give a confidence range over some interval. We could map that interval to the range we have here for Azure, but it may be better to have separate configuration options for each supported provider.